src/pages/register/index.js

```diff
@@ -6,49 +6,86 @@ import CredentialsCard from "../../components/credentials";
 import "./register.css";
 
 const Register = () => {
-	const { onRegister } = useAuth();
-	const [formData, setFormData] = useState({ email: "", password: "" });
+  const { onRegister } = useAuth();
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [errorMsg, setErrorMsg] = useState("");
+  const specialChars = /^(?=.*[!@#$%^&*])/;
+  const numbers = /^(?=.*[0-9])/;
+  // This doesn't work and could be implemented later
+  // const forbiddenChars = /^(?=.*[(){}[]|`¬¦!"£$%^&*"<>:;#~_-+=,@])/;
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
 
-	const onChange = (e) => {
-		const { name, value } = e.target;
-		setFormData({ ...formData, [name]: value });
-	};
+  function checkPasswordAndSubmit() {
+    let enteredPass = formData.password;
+    setErrorMsg("");
+    if (enteredPass.length < 8) {
+      setErrorMsg("Password must be at least 8 characters long!");
+      return false;
+    } else if (
+      enteredPass.toLowerCase() === enteredPass ||
+      enteredPass.toUpperCase() === enteredPass
+    ) {
+      setErrorMsg(
+        "Password must contain at least 1 uppercase and 1 lowercase letter"
+      );
+      return false;
+    } else if (!numbers.test(enteredPass)) {
+      setErrorMsg("Password must contain at least 1 number");
+      return false;
+    } else if (!specialChars.test(enteredPass)) {
+      setErrorMsg("Password must contain at least 1 special character");
+      return false;
+    }
+    // This does not work and should be fixed later
+    // if (forbiddenChars.test(enteredPass)) {
```

Comment:
remove comments for things that don't work - can add a bug / task on the backlog to then look at this separately, if more time available